src/ng/parse.js

```diff
@@ -1775,6 +1798,14 @@ function $ParseProvider() {
 
     function $parse(exp, interceptorFn) {
       var parsedExpression, oneTime, cacheKey;
+      var finalExposeAst = globalExposeAst;
```

Comment:
The `final` prefix is kind of confusing imo. I would just call it `exposeAst` (or `localExposeAst`).